logic/src/commonMain/kotlin/com/wire/kalium/logic/configuration/UserConfigRepository.kt

```diff
@@ -383,4 +389,16 @@ class UserConfigDataSource(
         wrapStorageRequest {
             userConfigDAO.setMigrationConfiguration(configuration.toEntity())
         }
+
+    override fun setLegalHoldRequest(
+        clientId: String,
+        lastPreKeyId: Int,
+        lastPreKey: String
+    ): Either<StorageFailure, Unit> = wrapStorageRequest {
+        userConfigStorage.persistLegalHoldRequest(clientId, lastPreKeyId, lastPreKey)
```

Comment:
can we please store this config value in the userConfigDAO for the following reasons:
1. userConfigDAO have its values cached so there is no fear of extra DB io
2. having a unified place for all of our config values
3. less usage of user scoped encrypted shared pref